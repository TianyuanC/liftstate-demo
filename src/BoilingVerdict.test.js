import React from "react";
import { BoilingVerdict } from "./BoilingVerdict";
import renderer from "react-test-renderer";

it("would not boil if no input", () => {
    const tree = renderer.create(<BoilingVerdict />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("would not boil if NaN", () => {
    const tree = renderer.create(<BoilingVerdict celsius={"nah"} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("would boil if at 100", () => {
    const tree = renderer.create(<BoilingVerdict celsius={100} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("would boil if above 100", () => {
    const tree = renderer.create(<BoilingVerdict celsius={200} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("would not boil if below 20", () => {
    const tree = renderer.create(<BoilingVerdict celsius={20} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it("would not boil if negative -1000", () => {
    const tree = renderer.create(<BoilingVerdict celsius={-1000} />).toJSON();
    expect(tree).toMatchSnapshot();
});
