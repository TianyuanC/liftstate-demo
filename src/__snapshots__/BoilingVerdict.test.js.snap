// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`would boil if above 100 1`] = `
<p>
  The water would boil.
</p>
`;

exports[`would boil if at 100 1`] = `
<p>
  The water would boil.
</p>
`;

exports[`would not boil if NaN 1`] = `
<p>
  The water would not boil.
</p>
`;

exports[`would not boil if below 20 1`] = `
<p>
  The water would not boil.
</p>
`;

exports[`would not boil if negative -1000 1`] = `
<p>
  The water would not boil.
</p>
`;

exports[`would not boil if no input 1`] = `
<p>
  The water would not boil.
</p>
`;
